SalesforceCommerceCloud/pwa-kit
# Allow generating test-project with correct Einstein's site id

# Description

For our testing, we use the **site id** `RefArchGlobal` for the Commerce API, while it's `aaij-MobileFirst` for Einstein. When we test https://pwa-kit.mobify-storefront.com/, we can see the Einstein calls are successful. However, if we test generating a  project with the preset `test-project`, the resulting project would show 400 errors for the Einstein calls because the Einstein's site id is not `aaij-MobileFirst`.

This PR fixes the above issue by allowing the test-project to have a different site id for Einstein.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

1. First generate a project locally by running this from the root: `GENERATOR_PRESET=test-project node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/project`
2. `cd /tmp/project`
3. Verify that the file 'app/api.config.js' has the correct site id `aaij-MobileFirst` for Einstein
3. `npm start`
4. Browse localhost:3000
5. Navigate to the PDP
6. Verify in the browser console that there is no error for any Einstein calls.


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
